contracts/RibbonStakingRewards.sol

```diff
@@ -157,7 +157,7 @@ contract StakingRewards is
     }
 
     function getReward() public override nonReentrant updateReward(msg.sender) {
-        uint256 reward = rewards[msg.sender];
+        uint256 reward = block.timestamp >= periodFinish ? rewards[msg.sender] : 0;
```

Comment:
quick one liner for locking solution. "locks" for entire month. if you withdraw beforehand you lose all rewards, but rewards are calculated based on how many weeks you were staking for.

Reply:
Just to clarify, this doesn't actually forfeit the user's rewards right? What this change does is to ensure when `getReward` is called, 0 RBN is returned. This is desirable.

One change I recommend we do, we should add `rewards[msg.sender] = 0` when the user calls `withdraw`. This basically forfeits the user's rewards when the user withdraws

```solidity
function withdraw(uint256 amount)
    public
    override
    nonReentrant
    updateReward(msg.sender)
{
    rewards[msg.sender] = 0;
    require(amount > 0, "Cannot withdraw 0");
    _totalSupply = _totalSupply.sub(amount);
    _balances[msg.sender] = _balances[msg.sender].sub(amount);
    stakingToken.safeTransfer(msg.sender, amount);
    emit Withdrawn(msg.sender, amount);
}
```

Reply:
yes, you are right. technically they can go back and forth, depositing friday before reward and withdrawing right after. they cannot get the rewards, but they can get them much later and at the amount as if they stayed the whole time